token/program-2022/src/state.rs

```diff
@@ -80,6 +127,53 @@ impl Pack for Mint {
     }
 }
 
+/// Type-Length-Value Entry, used to encapsulate all mixins contained within an account
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct TlvEntry<T, V> {
+    pub account_type: T,
```

Comment:
That's fair, I preferred to separate them mainly because the names were annoying as one enum, ie,
```
pub enum MixinType {
    MintTransferFee,
    AccountTransferFee,
    MintCloseAuthority,
}
```
And also, I thought it would be clearer / easier to reason in the code when iterating through the TLV entries.  I'd prefer to keep it for now, and if we need to combine later, it's not a huge deal.